### PR TITLE
8290264 :  java/util/concurrent/locks/Lock/OOMEInAQS.java fails with "exit code: 0"

### PR DESCRIPTION
**Pull request by DougLea**
This test now conforms to jtreg rules about not using System.exit to cover untested OutOfMemoryErrors

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8290264](https://bugs.openjdk.org/browse/JDK-8290264): java/util/concurrent/locks/Lock/OOMEInAQS.java fails with "exit code: 0"

### Reviewers
 * [David Holmes](https://openjdk.org/census#dholmes) (@dholmes-ora - **Reviewer**) ⚠️ Review applies to [971f3343](https://git.openjdk.org/jdk/pull/9491/files/971f3343224f4c6608d129e2255b3c1262ab70b6)
 * [Alan Bateman](https://openjdk.org/census#alanb) (@AlanBateman - **Reviewer**)
 * [Daniel D. Daugherty](https://openjdk.org/census#dcubed) (@dcubed-ojdk - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk pull/9491/head:pull/9491` \
`$ git checkout pull/9491`

Update a local copy of the PR: \
`$ git checkout pull/9491` \
`$ git pull https://git.openjdk.org/jdk pull/9491/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 9491`

View PR using the GUI difftool: \
`$ git pr show -t 9491`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/9491.diff">https://git.openjdk.org/jdk/pull/9491.diff</a>

</details>

### CODE REVIEW

**Review by dholmes-ora — APPROVED** [review]
Hi Doug,

One pre-existing style nit but otherwise the termination logic seems okay.

Thanks.

File: test/jdk/java/util/concurrent/locks/Lock/OOMEInAQS.java [hunk 1]
```diff
@@ -98,12 +107,17 @@ public void run() {
                 } finally {
                     lock.unlock();
```
[the remark this answers is not in the document]
Comment (DougLea) [reply test/jdk/java/util/concurrent/locks/Lock/OOMEInAQS.java:108]: Thanks.; changed

**Review by AlanBateman — APPROVED** [review]
Sorry I missed the use of System.exit when looking at the origin version. The update looks good (the explicit System.gc is probably not needed, setting data to null is enough)

**Review by dcubed-ojdk — APPROVED** [review]
Thumbs up.

**Comment by dcubed-ojdk** [comment]
I'm kicking off a Mach5 Tier1 on the latest version of this patch (v01).

**Pull request closed without merging**